docs/tutorial/file_handling.md

```diff
@@ -0,0 +1,471 @@
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      exists <- FilesOs.exists(annoyingFile)
+      _ <- if exists then FilesOs.delete(annoyingFile)
+           else IO.unit
+    yield ()
+
+end Deleting
+```
+
+@:@
+
+Note that we first check if the file exists before deleting it, this is because trying to delete a file with `delete` that does not exist will result in an error. If that is not your style, you have two options. One is using the `whenA` combinator from `cats.syntax.applicative.*`:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.syntax.applicative.* // You can instead import cats.syntax.all.* !
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      exists <- annoyingFile.exists
+      _ <- annoyingFile.delete.whenA(exists)
+    yield ()
+
+end Deleting
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.syntax.applicative.* // You can instead import cats.syntax.all.* !
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      exists <- FilesOs.exists(annoyingFile)
+      _ <- FilesOs.delete(annoyingFile).whenA(exists)
+    yield ()
+
+end Deleting
+```
+
+@:@
+
+Or even better, use the convenience method `deleteIfExists`:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      deleted <- annoyingFile.deleteIfExists
+      _ <- IO.println(s"Are they reaching out? $deleted")
+    yield ()
+
+end Deleting
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      deleted <- FilesOs.deleteIfExists(annoyingFile)
+      _ <- IO.println(s"Are they reaching out? $deleted")
+    yield ()
+
+end Deleting
+```
+
+@:@
+
+It will return a boolean indicating whether the file has been deleted (also deletes directories).
+
+Lastly, you may want to delete not one but multiple files and directories, here is when the `deleteDirectorires` comes handy, as it will delete all the files and directories recursively (similar to `rm -r`):
```

Comment:
```suggestion
Finally, you may want to delete not one but multiple files and directories, here is when the `deleteDirectorires` comes handy, as it will delete all the files and directories recursively (similar to `rm -r`):
```